### ISSUE TITLE
Nuxt-rc.8 prefetches all pages (ssr: false)

### ISSUE THREAD

**Issue by stenet** [nuxt/framework#6593]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.7.0`
- Nuxt Version:     `3.0.0-rc.8`
- Package Manager:  `npm@8.17.0`
- Builder:          `vite`
- User Config:      `ssr`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

You can clone https://github.com/stenet/nuxt-many-pages and run

```
nuxt build
```

to see the result (.output/public/index.html) with Nuxt-rc.6. After that update to Nuxt-rc.8 and rebuild again. You should see the massive increase.

### Describe the bug

I did a few tests with a lot of pages.

In Nuxt-rc.6, after building the app, the .output/public/index.html had only the entry.xxx.mjs as script and a total size of about 300 bytes.

After upgrading to Nuxt-rc.8, the index.html has every page in the app as as link prefetch script and a total size of about 87 kilobytes. Btw the entry.xxx.mjs is added twice (1x as script + 1x as link).

Also I understand this from the point of better performance when navigating/loading such a page the first time, but I think that this behavior is rather suboptimal when the app has many pages of which the user calls only a few of them (or has no access) and the browser cache is "filled up" with a lot of unnecessary content.

**Comment by dikesh** [comment]
Facing the same issue while prerendering pages with `nuxt generate` . All the other pages are getting prefetched as script.

E.g. I have three pages `/`, `/children/child1` and `/children2/child2` and when I load `/`, the page loads `.mjs` scripts for `child1` and `child2` as well.

![image](https://user-images.githubusercontent.com/1618526/184630765-38cda06b-9934-47fe-9cdb-61e730cf71ec.png)

Code: https://stackblitz.com/edit/nuxt-starter-pasbl6

**Comment by pi0** [comment]
ping @danielroe

**Comment by danielroe** [comment]
This is currently the correct behaviour. Prefetching is low priority and takes place in browser idle time. Although the network requests might look alarming in the network tab, they are non-blocking and should not affect user experience.

After merging https://github.com/nuxt/framework/issues/3649, I think we can likely revisit this behaviour.

Also note that [we removed](https://github.com/nuxt/framework/pull/6210#discussion_r937950236) the ability for users to customise `shouldPreload` and `shouldPrefetch`, but this will be made possible, either at build-time or runtime.

**Comment by pi0** [comment]
I believe default prefetching behavior for SPA should be non when we cannot predict sibling routes this can grow infinity and indeed later we can enable other prefetching mechanisms such as relying on links and intersection observer...

**Comment by danielroe** [comment]
Currently dynamic imports include things like routes, error components, and any global components (so if using with `@nuxt/content`). I am _very_ concerned to decrease HTML and number of prefetched resources but if we simply disable prefetching, sites will seem much slower.

From my point of view, before removing current prefetch behaviour, we need the ability to make some finer-grained decisions.

A set of (ordered) suggestions:

1. we can merge `prefetch` support via NuxtLink (https://github.com/nuxt/framework/pull/4329 currently awaiting review from @pi0)
2. we can remove page chunks from dynamic imports in bundle renderer
3. we can support programmatic prefetching of global components
4. (optionally) add a plugin to update manifest so that the components that _might_ use global components are the ones that have the dynamic import registered

**Comment by pi0** [comment]
I think we can do 1 and 2 in parallel until next RC. Programmatic API to prefetch components also sound like a good idea to be supported as a helper util 👍🏼

**Comment by danielroe** [comment]
Great! Let's track in https://github.com/nuxt/framework/issues/6638.
